pkg/scyllaclient/client_rclone.go

```diff
@@ -460,22 +462,43 @@ func (c *Client) RcloneListDirIter(ctx context.Context, host, remotePath string,
 		}
 	}
 
-	var v RcloneListDirItem
-	for dec.More() {
-		// Detect context cancellation
-		if ctx.Err() != nil {
-			return ctx.Err()
+	errCh := make(chan error)
+	go func(ch chan error) {
+		var v RcloneListDirItem
+		for dec.More() {
+			// Detect context cancellation
+			if ctx.Err() != nil {
+				ch <- ctx.Err()
+				break
+			}
+
+			// Read value
+			v = RcloneListDirItem{}
+			if err := dec.Decode(&v); err != nil {
+				ch <- err
+			}
+			f(&v)
+
+			ch <- nil
 		}
-
-		// Read value
-		v = RcloneListDirItem{}
-		if err := dec.Decode(&v); err != nil {
-			return err
+		close(ch)
+	}(errCh)
```

Comment:
We use ctx from the outer scope why pass errCh as ch?
It makes things less readable with no clear goal.